autograd/core.py

```diff
@@ -94,42 +92,35 @@ def __init__(self, vs, mut_add):
 def zero_vjp(argnum):
     return lambda ans, *args, **kwargs: lambda g: vspace(args[argnum]).zeros()
 
-primitive_vjps = defaultdict(dict)
+primitive_vjps_onearg = defaultdict(dict)
+primitive_vjps = {}
 
-def primitive_vjp(fun, argnum, ans, args, kwargs):
-    try:
-        vjp = primitive_vjps[fun][argnum]
-    except KeyError:
-        if primitive_vjps[fun]:
-            errstr = "Gradient of {0} w.r.t. arg number {1} not yet implemented."
-        else:
-            errstr = "Gradient of {0} not yet implemented."
-        raise NotImplementedError(errstr.format(repr(fun), argnum))
-    return vjp(ans, args, kwargs)
+def defvjp_argnums(fun, vjpmaker):
+    primitive_vjps[fun] = vjpmaker
 
 def defvjp(fun, vjpmaker, argnum=0):
-    def vjp_fixed_args(ans, args, kwargs):
-        return vjpmaker(ans, *args, **kwargs)
-    primitive_vjps[fun][argnum] = vjp_fixed_args
+    primitive_vjps_onearg[fun][argnum] = vjpmaker
+    def vjp_argnums(argnums, ans, args, kwargs):
+        vjps_dict = primitive_vjps_onearg[fun]
+        vjps = [vjps_dict[argnum](ans, *args, **kwargs) for argnum in argnums]
```

Comment:
This could be a generator expression (with round brackets instead of square), though that doesn't seem to make any difference to performance on my machine.

Reply:
The problem with making it a generator expression is that it then gets consumed once it's used. That's a problem any time you want to call `make_vjp(f)(x)` more than once, which happens in `jacobian` and elsewhere.